### PR TITLE
updated release notes for maintainers

### PR DESCRIPTION
**Pull request by jspaaks**
**List of related issues or pull requests**

Refs: -
**Describe the changes made in this pull request**

Updated release notes for maintainers in README.dev.rst

**Instructions to review the pull request**

look at [README.dev.rst](https://github.com/fair-software/howfairis/blob/readme-dev-release-notes/README.dev.rst)

### CODE REVIEW

File: README.dev.rst [hunk 1]
```diff
@@ -89,13 +90,25 @@ Bumping the version across all files is done with bump2version, e.g.
...
 
 Making a release
-^^^^^^^^^^^^^^^^
+----------------
 
-Make sure the version is correct.
+Preparation
+^^^^^^^^^^^
+
+1. Write the ``CHANGELOG.md``
```
Comment (fdiblen) [inline README.dev.rst:98]:
```suggestion
1. Update the ``CHANGELOG.md``
```

File: README.dev.rst [hunk 2]
```diff
@@ -89,13 +90,25 @@ Bumping the version across all files is done with bump2version, e.g.
...
+2. Verify that the information in ``CITATION.cff`` is correct, and that ``.zenodo.json`` contains equivalent data
+3. Make sure the version has been updated.
+4. Run the unit tests with ``pytest tests/``
+5. Run the live tests with ``pytest livetests/``
+6. Run the clitests with ``bash clitests/script.sh``
+
```
Comment (fdiblen) [inline README.dev.rst:104]:
```suggestion
1. Make sure all the GitHub actions succeed
```

**Review by sverhoeven — APPROVED** [review]
LGTM

Will create issue for Docker permission/versioning

**Review by fdiblen — APPROVED** [review]
I aprove @sverhoeven's approval. :ok:

**Comment by jspaaks** [comment]
Thanks guys!

**Pull request merged**